app/services/authentication/authenticator.rb

```diff
@@ -143,14 +144,26 @@ def update_profile_updated_at(user)
 
     def account_less_than_a_week_old?(user, logged_in_identity)
       provider_created_at = user.public_send(provider.user_created_at_field)
-      user_identity_age = provider_created_at ||
-        Time.zone.parse(logged_in_identity.auth_data_dump.extra.raw_info.created_at)
+      user_identity_age = provider_created_at
+      user_identity_age ||= extract_created_at_from_payload(logged_in_identity)
 
       # last one is a fallback in case both are nil
       range = 1.week.ago.beginning_of_day..Time.current
       range.cover?(user_identity_age)
     end
 
+    def extract_created_at_from_payload(logged_in_identity)
+      raw_info = logged_in_identity.auth_data_dump.extra.raw_info
+
+      if raw_info.created_at.present?
+        Time.zone.parse(raw_info.created_at)
+      elsif raw_info.auth_time.present?
```

Comment:
Did we find some other edge cases that caused these additional booleans here?

Reply:
I don't think so, as in I never ran into another odd scenario. In this case the `else` handles when Providers (Apple is the one I can recall) don't provide the complete data in their payload which then uses the `Time.current` default